sql/plan/join_iters.go

```diff
@@ -0,0 +1,612 @@
+// Copyright 2020-2021 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package plan
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"reflect"
+
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/trace"
+
+	"github.com/dolthub/go-mysql-server/sql"
+)
+
+func newJoinIter(ctx *sql.Context, j *JoinNode, row sql.Row) (sql.RowIter, error) {
+	var leftName, rightName string
+	if leftTable, ok := j.left.(sql.Nameable); ok {
+		leftName = leftTable.Name()
+	} else {
+		leftName = reflect.TypeOf(j.left).String()
+	}
+
+	if rightTable, ok := j.right.(sql.Nameable); ok {
+		rightName = rightTable.Name()
+	} else {
+		rightName = reflect.TypeOf(j.right).String()
+	}
+
+	span, ctx := ctx.Span("plan.indexedJoin", trace.WithAttributes(
```

Comment:
nit: maybe not such an appropriate name...just `plan.joinIter` span or something?